ocean_provider/utils/basics.py

```diff
@@ -17,14 +18,36 @@
 from web3.main import Web3
 
 
-def get_config():
-    config_file = os.getenv("CONFIG_FILE", "config.ini")
-    return Config(filename=config_file)
+def get_config(config_file: Optional[str] = None) -> Config:
+    config_file = config_file if config_file else os.getenv("CONFIG_FILE", "config.ini")
 
+    return (
+        Config(filename=config_file)
+        if config_file
+        else os.getenv("CONFIG_FILE", "config.ini")
```

Comment:
I just realized the code snippet I recommended in my last set of comments had a bug in it - the else returns the `CONFIG_FILE` string without building a `Config` instance :facepalm: So this code needs to be fixed.

Also, you're currently using 2 ternary operators but it can be done rather succinctly with just 1.

Here's my new suggestion - this time with no bugs! (hopefully, :crossed_fingers: )

```python
def get_config(config_file: Optional[str] = None) -> Config:
    return Config(
        filename=config_file
        if config_file is not None
        else os.getenv("CONFIG_FILE", "config.ini")
    )
```